narwhal/worker/src/worker.rs

```diff
@@ -177,6 +188,10 @@ impl Worker {
                 inbound_network_metrics,
             )))
             .layer(CallbackLayer::new(FailpointsMakeCallbackHandler::new()))
+            .layer(SetRequestHeaderLayer::overriding(
```

Comment:
Similar question as above, is this needed here?

Reply:
This was recommended by Brandon for debugging